src/thermostat.js

```diff
@@ -17,6 +19,42 @@ export default function createThermostat({ Service, Characteristic }) {
       this.thermostatService = new Service.Thermostat(this.name)
       this.informationService = new Service.AccessoryInformation()
       this.boostService = new Service.Switch(`${this.name} boost mode`)
+      this.currentTemperature = null
+
+      if (config.currentTemperature) {
+        this.mqttClient = mqtt.connect(config.currentTemperature.url, {
+          keepalive: 10,
+          clientId: 'mqttjs_'.concat(Math.random().toString(16).substr(2, 8)),
+          protocolId: 'MQTT',
+          protocolVersion: 4,
+          clean: true,
+          reconnectPeriod: 1000,
+          connectTimeout: 30 * 1000,
+          will: {
+            topic: 'WillMsg',
+            payload: 'Connection Closed abnormally..!',
+            qos: 0,
+            retain: false,
+          },
+          username: config.currentTemperature.username,
+          password: config.currentTemperature.password,
+          rejectUnauthorized: false,
+        })
+
+        const that = this
```

Comment:
Please remove this line and simply use `this` in the lines below.
You don't have do use a reference for `this` as you're using arrow functions :)